addons/mail/models/discuss/discuss_channel.py

```diff
@@ -1021,6 +1022,13 @@ def channel_pin(self, pinned=False):
         else:
             self.env['bus.bus']._sendone(self.env.user.partner_id, 'mail.record/insert', {"Thread": self._channel_info()[0]})
 
+    def _fetch_last_message(self, extra_domain=()):
```

Comment:
Generally speaking, `extra_domain` is not safe allows unforeseen domain change, if we just want an option for id to be smaller than a value, we should just allow that as param.

Reply:
Because `_get_last_messages` is not good from a SQL PoV.
The fact that it does a `MAX` means that it will need to still fetch and aggregate all the messages on the SQL server, even though the information is already in the index.

Reply:
Are you talking from a security PoV?
If so, using `expression.AND` should make it safe (also the fact that the extra domain is appended and not prepended)